### PR TITLE
Make O_TRUNC use better

### PR DESCRIPTION
**Pull request by tempusfrangit**
Only use if --force is actually specified

### CODE REVIEW

**Comment by tempusfrangit** [comment]
This did reduce the delta (minimally) between tar and non-tar mode. I'm not certain this is worth it. @philandstuff thoughts?

File: pkg/consumer/consumer.go [hunk 1]
```diff
@@ -4,4 +4,5 @@ import "io"
 
 type Consumer interface {
 	Consume(reader io.Reader, destPath string) error
+	SetOverwriteTarget(force bool)
```
Comment (philandstuff) [inline pkg/consumer/consumer.go:7]: eeh, I don't like this interface. We should pass a slice/varargs of ConsumeOptions (of which OverwriteTarget would be one instance) into Consume() instead.

**Comment by philandstuff** [comment]
>  This did reduce the delta (minimally) between tar and non-tar mode. I'm not certain this is worth it. @philandstuff thoughts?

I think if we can tidy up the interface then this is a good idea. I'd also like to see us do the partial file download thing, where we download to a temp file and move to the target once the file is fully written

**Comment by tempusfrangit** [comment]
Superseded by #158

**Pull request closed without merging**